src/mscorlib/src/System/Threading/SpinLock.cs

```diff
@@ -339,30 +339,30 @@ private void ContinueTryEnter(int millisecondsTimeout, ref bool lockTaken)
 
                 if (Interlocked.CompareExchange(ref m_owner, observedOwner | 1, observedOwner, ref lockTaken) == observedOwner)
                 {
+                    // Aquired lock
                     return;
                 }
 
 #if !FEATURE_CORECLR
                 Thread.EndCriticalRegion();
 #endif
+                if (millisecondsTimeout == 0)
```

Comment:
Probably rather than attempting a double acquire. It would mean adding another ` Thread.EndCriticalRegion()` on a not acquire branch. 

However its a pretty heavily stacked function which does currently always inline; maybe revisit as an enhancement?